### PR TITLE
clean_blood refactor

### PR DESCRIPTION
**Pull request by CornMyCob**
## What Does This PR Do
Refactors clean_blood code so that showers and space cleaner don't use two different bits of code to do exactly the same thing.

## Why It's Good For The Game
Code consistency.

### CODE REVIEW

File: code/game/objects/structures/watercloset.dm [hunk 1]
```diff
@@ -403,75 +403,7 @@
...
 		to_chat(L, "<span class='warning'>You've been drenched in water!</span>")
-		if(iscarbon(O))
-			var/mob/living/carbon/M = O
-			if(M.r_hand)
-				M.r_hand.clean_blood()
-			if(M.l_hand)
-				M.l_hand.clean_blood()
-			if(M.back)
-				if(M.back.clean_blood())
-					M.update_inv_back()
```
Comment (SteelSlayer) [inline code/game/objects/structures/watercloset.dm:414]: You got rid of `back.clean_blood()` here but I don't see it added into `/human/clean_blood()`, which means backpacks won't get cleaned. Is that intended?

Comment (CornMyCob) [reply code/game/objects/structures/watercloset.dm:414]: Nope, not intended. Thanks nerd.

File: code/game/atoms.dm [hunk 2]
```diff
@@ -701,16 +701,56 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
...
+	if(head)
+		if(head.clean_blood())
+			update_inv_head()
+	if(wear_suit)
+		if(wear_suit.clean_blood())
+			update_inv_wear_suit()
+	else if(w_uniform)
+		if(w_uniform.clean_blood())
+			update_inv_w_uniform()
+	if(!(wear_suit && wear_suit.flags_inv & HIDEGLOVES))
```
Comment (Fox-McCloud) [inline code/game/atoms.dm:714]: I'd much prefer the styling of:

`if(wear_suit && !(wear_suit.flags_inv & HIDEGLOVES))`

File: code/game/atoms.dm [hunk 3]
```diff
@@ -701,16 +701,56 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
...
+		if(gloves)
+			if(gloves.clean_blood())
+				update_inv_gloves()
+				gloves.germ_level = 0
+		else if(bloody_hands)
 			bloody_hands = 0
 			update_inv_gloves()
+	if(!(wear_suit && wear_suit.flags_inv & HIDESHOES))
```
Comment (Fox-McCloud) [inline code/game/atoms.dm:722]: same

File: code/game/atoms.dm [hunk 4]
```diff
@@ -701,16 +701,56 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
...
+		if(shoes)
+			if(shoes.clean_blood())
+				update_inv_shoes()
+		else
+			feet_blood_color = null
+			qdel(feet_blood_DNA)
+			bloody_feet = list(BLOOD_STATE_HUMAN = 0, BLOOD_STATE_XENO = 0,  BLOOD_STATE_NOT_BLOODY = 0)
+			blood_state = BLOOD_STATE_NOT_BLOODY
+			update_inv_shoes()
+	if(!(head && head.flags_inv & HIDEMASK))
```
Comment (Fox-McCloud) [inline code/game/atoms.dm:732]: Again.

File: code/game/objects/structures/watercloset.dm [hunk 5]
```diff
@@ -403,75 +403,7 @@
 		L.ExtinguishMob()
 		L.adjust_fire_stacks(-20) //Douse ourselves with water to avoid fire more easily
 		to_chat(L, "<span class='warning'>You've been drenched in water!</span>")
-		if(iscarbon(O))
-			var/mob/living/carbon/M = O
-			if(M.r_hand)
-				M.r_hand.clean_blood()
-			if(M.l_hand)
-				M.l_hand.clean_blood()
```
Comment (Fox-McCloud) [inline code/game/objects/structures/watercloset.dm:411]: The above two aren't put in anywhere else.

**Comment by Fox-McCloud** [comment]
Very nice clean up.

Though, one change you should make to this: some of these holder vars are defined at the carbon level; those clothing/equipable slots that are at a particular defined level should be at that level's `clean_blood`---so, for example, Xenos with blood larva in their claws/paws/whatever they're called get cleaned of blood if they get hit by space cleaner or go into a shower.

File: code/game/atoms.dm [hunk 6]
```diff
@@ -701,16 +701,40 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
 
 
 /mob/living/carbon/human/clean_blood()
```
Comment (Fox-McCloud) [inline code/game/atoms.dm:703]: inconsistent args.

File: code/modules/mob/living/carbon/carbon.dm [hunk 7]
```diff
@@ -1201,3 +1201,22 @@ so that different stomachs can handle things in different ways VB*/
 		I.acid_level = 0 //washes off the acid on our clothes
 		I.extinguish() //extinguishes our clothes
 	..()
+
+/mob/living/carbon/clean_blood(clean_hands = TRUE, clean_mask = TRUE, clean_feet = TRUE)
+	// Declared vars because these get changed around
+	var/do_clean_hands = clean_hands
+	var/do_clean_mask = clean_mask
+	var/do_clean_feet = clean_feet
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/carbon.dm:1209]: you don't need these if you're going to have args.

File: code/game/atoms.dm [hunk 8]
```diff
@@ -701,16 +701,40 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
...
-	if(gloves)
+	var/do_clean_hands = TRUE
+	var/do_clean_mask = TRUE
+	var/do_clean_feet = TRUE
```
Comment (Fox-McCloud) [inline code/game/atoms.dm:706]: Just use args to do this.

File: code/modules/mob/mob.dm [hunk 9]
```diff
@@ -1390,3 +1390,29 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
 ///Force set the mob nutrition
 /mob/proc/set_nutrition(change)
 	nutrition = max(0, change)
+
+/mob/clean_blood(clean_hands = TRUE, clean_mask = TRUE, clean_feet = TRUE)
+	. = ..()
+	if(bloody_hands && clean_hands)
+		bloody_hands = 0
```
Comment (AffectedArc07) [inline code/modules/mob/mob.dm:1397]: FALSE here

File: code/modules/mob/mob.dm [hunk 10]
```diff
@@ -1390,3 +1390,29 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
...
+	if(back)
+		if(back.clean_blood())
+			update_inv_back()
+	if(wear_mask && clean_mask)
+		if(wear_mask.clean_blood())
+			update_inv_wear_mask()
+	if(clean_feet)
+		feet_blood_color = null
+		qdel(feet_blood_DNA)
+		bloody_feet = list(BLOOD_STATE_HUMAN = 0, BLOOD_STATE_XENO = 0,  BLOOD_STATE_NOT_BLOODY = 0)
```
Comment (AffectedArc07) [inline code/modules/mob/mob.dm:1414]: Should the zeroes here be `FALSE`

File: code/modules/mob/mob.dm (hunk 9 above)
Comment (CornMyCob) [reply code/modules/mob/mob.dm:1397]: Not a bool

File: code/modules/mob/mob.dm (hunk 10 above)
Comment (CornMyCob) [reply code/modules/mob/mob.dm:1414]: No they shouldn't

File: code/modules/mob/mob.dm (hunk 9 above)
Comment (AffectedArc07) [reply code/modules/mob/mob.dm:1397]: TIL this game has the worst variable naming conventions

**Pull request merged**